### DISCUSSION TITLE
Not able to resize on using layouts

### DISCUSSION THREAD

**Discussion by SParvathySuresh** [eclipse-glsp/glsp#774] [Q&A] [answered]
I am using HBOX layout for a rectangular node that has both editlabel feature and resize feature. Resize handles are shown for the node. But, on dragging/adjusting the size manually, the node is not getting resized. It is taking the label width always. May I know  what I might be missing. I have added below lines under setProperties of the nodebuilder.java file. Also have enabled resize feature for the node from diconfig file.

node.setLayout(GConstants.Layout.HBOX);
node.getLayoutOptions().put("paddingRight", 10);
node.getLayoutOptions().put("resizeContainer", true);
node.getLayoutOptions().put("minWidth", 128);
node.getChildren().add(createCompartmentHeader(node));

**Comment by CamilleLetavernier** [comment, marked as answer]
Hi,

To resize a node, enabling the "resizable" feature is only the first step: it enables the interaction on the client side, but it still needs to be handled by the server as well.

The server should:

- Handle the ChangeBoundsOperation and modify the prefWidth/prefHeight of the node (This should be done by the default handler already; just make sure you use it, or override it in a way that still respects this behavior)
- Ensure the prefWidth/prefHeight is set in the layout options of the HBox Node before it is sent back to the client (If you directly serialize the Json Node as we do in the Workflow Example, this should already be covered by the ChangeBoundsOperationHandler. If you're using your own serialization and then deriving the Json Graph at runtime, then you need to make sure the prefWidth/prefHeight layout options are set again on each request).
